### PR TITLE
Implement `to(torch.device)` for Ragged<T>

### LINKED ISSUE #367: .to() for Ragged
Can someone please add a to(device: torch.Device) function to class Ragged's binding code?
Need it for something, probably by to-morrow afternoon.

### PR DESCRIPTION
**Pull request by csukuangfj**
Closes #367 

fsa.py is changed slightly. 

Hope that it will not make much trouble for you @danpovey

### CODE REVIEW

**Comment by qindazhu** [comment]
+1

File: k2/python/csrc/torch/ragged.cu [hunk 1]
```diff
@@ -24,6 +24,22 @@ static void PybindRaggedTpl(py::module &m, const char *name) {
...
+    K2_CHECK(device_type == "cpu" || device_type == "cuda")
+        << "Unsupported device type: " << device_type;
+
+    ContextPtr &context = self.Context();
+    if (device_type == "cpu") {
+      if (context->GetDeviceType() == kCpu) return self;
+      return self.To(GetCpuContext());
+    }
+
+    if (context->GetDeviceType() == kCuda) return self;
```
Comment (danpovey) [inline k2/python/csrc/torch/ragged.cu:38]: You can't assume there is just one CUDA device!

Comment (csukuangfj) [reply k2/python/csrc/torch/ragged.cu]: You're right. Fixed.

**Comment by danpovey** [comment]
Thanks!

**Pull request merged**